channelserver/src/ip_rate_limit.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::HashMap;
+use std::time::Duration;
+
+use reqwest::{self, header};
+use serde_json::Value;
+
+use perror::{HandlerError, HandlerErrorKind};
+use settings::Settings;
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct IPReputation {
+    iprepd_server: Option<String>, //IP Reputation server
+    iprep_min: u8,                 //Minimal IP reputation to accept.
+    iprep_violation: String,       //Violation to report
+}
+
+impl<'a> From<&'a Settings> for IPReputation {
+    fn from(settings: &'a Settings) -> Self {
+        let server = if settings.ip_abuse_server.len() > 0 {
+            Some(settings.ip_abuse_server.clone())
+        } else {
+            None
+        };
+        IPReputation {
+            iprepd_server: server,
+            iprep_min: settings.iprep_min,
+            iprep_violation: settings.ip_violation.clone(),
+            // TODO: add logger
+        }
+    }
+}
+
+impl IPReputation {
+    pub fn is_abusive(&self, addr: &str) -> Result<bool, HandlerError> {
+        if let Some(srv) = &self.iprepd_server {
+            // Check the Ops IPReputation server
+            let client = reqwest::Client::builder()
+                .timeout(Duration::from_secs(3))
+                .build()
+                .map_err(|err| {
+                    return HandlerErrorKind::InternalServerError(format!(
+                        "Could not build request client"
+                    ));
+                });
+            // https://github.com/mozilla-services/iprepd
+            let response = reqwest::get(&format!("https://{}/{}", srv, addr))
```

Comment:
IIUC this is going to make a blocking HTTP request, will that be a problem in the context of actix framework?  I wonder if we should use actix's own http client for better integration: https://actix.rs/actix-web/actix_web/client/index.html

Reply:
Using the actix request framework would make this less blocking. This was quicker (and frankly easier) to get going. 

I'll see if I can get a reasonable future chain that does what I need.